aws/rust-runtime/aws-http/src/content_encoding.rs

```diff
@@ -276,6 +282,28 @@ where
     }
 }
 
+/// Errors related to `AwsChunkedBody`
+#[derive(Debug)]
+enum AwsChunkedBodyError {
+    /// Error that occurs when the sum of `trailer_lengths` set when creating an `AwsChunkedBody` is
+    /// not equal to the actual length of the trailers returned by the inner `http::Body`
+    /// implementor. These trailer lengths are necessary in order to correctly calculate the total
+    /// size of the body for setting the content length header.
+    ReportedTrailerLengthMismatch { actual: u64, expected: u64 },
```

Comment:
I see @rcoh suggested making an error for this, but I think this is a case where we have a bug in our code, and there's little to no value for a customer in matching on this error. It's a case that simply should not happen if the code is correct, so an assertion was warranted.

On the flip side, with an assertion, it would silently allow the lengths to mismatch when compiled in release mode, which would likely result in a server-side error response. I think this is OK though, since a bug with this should not have been released in the first place if it was adequately tested.

We might want to proptest header names/values for the size prediction vs. the rendering to be absolutely certain they cannot mismatch. I think they can right now in the case where a header value has Unicode characters in it since the length predictor function is using string length instead of byte length.

Reply:
Another thing we could consider is just getting rid of the length prediction entirely, just render the trailers to a `BytesMut` when the length is initially needed, and store that `BytesMut` in the `AwsChunkedBody` for later.

Reply:
@jdisanti only `debug_assert` doesn't fail in release mode

Reply:
I added a new error for when stream_length is set incorrectly